### ISSUE TITLE
[macos] Exception: Podfile missing

### ISSUE THREAD

**Issue by YowFung** [flutter/flutter#95236]
I am writing my application for the desktop platform (macOS). Everything was normal before. But I don't know when to start, the `Podfile` file was missing.

```txt
Launching lib/main.dart on macOS in debug mode...
Exception: Podfile missing
```

- I tried to run `flutter create --platforms macos .`. It is unuseful.
- I deleted the `macos` folder and re-tried to run `flutter create --platforms macos .`. Also unuseful.
- I deleted the `macos` folder, ran `flutter clean && flutter pub get` and re-tried to run `flutter create --platforms macos`. Same issue.
- I tried to copy a `Podfile` file from another project here. It throws the following errors:

```txt
Launching lib/main.dart on macOS in debug mode...
Building macOS application...
--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:macOS, arch:x86_64, id:55A4053D-D81E-5CA4-9BE4-F613FB53F459 }
{ platform:macOS, name:Any Mac }
note: Using new build system
note: Planning
note: Build preparation complete
note: Building targets in parallel
warning: Unable to read contents of XCFileList '/Target Support Files/Pods-Runner/Pods-Runner-frameworks-Debug-output-files.xcfilelist' (in target 'Runner' from project 'Runner')
error: Unable to load contents of file list: '/Target Support Files/Pods-Runner/Pods-Runner-frameworks-Debug-input-files.xcfilelist' (in target 'Runner' from project 'Runner')
error: Unable to load contents of file list: '/Target Support Files/Pods-Runner/Pods-Runner-frameworks-Debug-output-files.xcfilelist' (in target 'Runner' from project 'Runner')
** BUILD FAILED **

Exception: Build process failed
```

So, how should I fix this problem?

<details>
<summary>flutter doctor -v</summary>

```txt
[✓] Flutter (Channel stable, 2.8.0, on macOS 12.0.1 21A559 darwin-x64, locale zh-Hans-CN)
    • Flutter version 2.8.0 at /Users/yowfung/Library/Flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision cf44000065 (5 days ago), 2021-12-08 14:06:50 -0800
    • Engine revision 40a99c5951
    • Dart version 2.15.0
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/yowfung/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/yowfung/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] IntelliJ IDEA Community Edition (version 2021.2.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 63.0.3
    • Dart plugin version 212.5632

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.25.0

[✓] Proxy Configuration
    • HTTP_PROXY is set
    • NO_PROXY is
      localhost,0.0.0.0,127.0.0.1,192.168.1.253,192.168.1.250,192.168.1.251,192.168.5.253,192.168.0.
      122,192.168.*,169.254.1.1,169.254.*
    • NO_PROXY contains 127.0.0.1
    • NO_PROXY contains localhost

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 12.0.1 21A559 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 96.0.4664.93

• No issues found!
```
</details>

**Comment by YowFung** [comment]
I tried to create a new project in another directory. It also has no `Podfile` file.

```txt
flutter create --platforms macos --platforms windows demo
```

![image](https://user-images.githubusercontent.com/26346560/145942529-d0cfde4f-8068-4047-9b83-01cf33902b9f.png)

**Comment by danagbemava-nc** [comment]
Hi @YowFung, to the best of my knowledge, a `Podfile` is only generated when you have added a plugin so it wouldn't be there in a new project. The project should still run regardless. 

That aside, can you provide your `pubspec.yaml` and `pubspec.lock` and `flutter run -v` of the project that is throwing the `Exception: Podfile missing`?

Thank you

**Comment by YowFung** [comment]
I have just discovered that this problem only appeared when I used a third-party plug-in. 

But why doesn't it automatically create the `Podfile` file? I have tried many plug-ins, and so have they.

my dependencies: 
```yaml
dependencies:
  flutter:
    sdk: flutter

  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^1.0.2
  get: ^4.6.1
  dio: ^4.0.4
  oauth2: ^2.0.0
  url_launcher: ^6.0.17
```
When I remove `url_launcher `, it works normally.

**Comment by danagbemava-nc** [comment]
Can you check if you still have the issue with `url_launcher` with a new project? 
I added it to a new project and the `Podfile` was successfully generated. See video below

<details>
<summary>video</summary>

https://user-images.githubusercontent.com/88313112/145986165-89109e54-744f-46a9-bb1b-a12387046558.mov

</details>

**Comment by YowFung** [comment]
It was really weird when I tried it in a new project. I tried many times yesterday and it not been generated, but today it was generated.

https://user-images.githubusercontent.com/26346560/146105761-256fcef1-1242-4100-bfe1-71a155ad0e6f.mov

However, on the application I am developing, this problem still exists.

https://user-images.githubusercontent.com/26346560/146105693-9c51aacb-ac83-40ed-8441-5565343d9c03.mov

**Comment by danagbemava-nc** [comment]
Can you share the `pubspec.lock` file?

**Comment by YowFung** [comment]
<details>
<summary>pubspec.lock</summary>

```yaml
# Generated by pub
# See https://dart.dev/tools/pub/glossary#lockfile
packages:
  async:
    dependency: transitive
    description:
      name: async
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "2.8.2"
  boolean_selector:
    dependency: transitive
    description:
      name: boolean_selector
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "2.1.0"
  characters:
    dependency: transitive
    description:
      name: characters
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "1.2.0"
  charcode:
    dependency: transitive
    description:
      name: charcode
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "1.3.1"
  clock:
    dependency: transitive
    description:
      name: clock
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "1.1.0"
  collection:
    dependency: transitive
    description:
      name: collection
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "1.15.0"
  crypto:
    dependency: transitive
    description:
      name: crypto
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "3.0.1"
  cupertino_icons:
    dependency: "direct main"
    description:
      name: cupertino_icons
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "1.0.4"
  dio:
    dependency: "direct main"
    description:
      name: dio
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "4.0.4"
  fake_async:
    dependency: transitive
    description:
      name: fake_async
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "1.2.0"
  ffi:
    dependency: transitive
    description:
      name: ffi
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "1.1.2"
  file:
    dependency: transitive
    description:
      name: file
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "6.1.2"
  flutter:
    dependency: "direct main"
    description: flutter
    source: sdk
    version: "0.0.0"
  flutter_lints:
    dependency: "direct dev"
    description:
      name: flutter_lints
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "1.0.4"
  flutter_test:
    dependency: "direct dev"
    description: flutter
    source: sdk
    version: "0.0.0"
  flutter_web_plugins:
    dependency: transitive
    description: flutter
    source: sdk
    version: "0.0.0"
  get:
    dependency: "direct main"
    description:
      name: get
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "4.6.1"
  http:
    dependency: transitive
    description:
      name: http
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "0.13.4"
  http_parser:
    dependency: transitive
    description:
      name: http_parser
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "4.0.0"
  js:
    dependency: transitive
    description:
      name: js
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "0.6.3"
  lints:
    dependency: transitive
    description:
      name: lints
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "1.0.1"
  matcher:
    dependency: transitive
    description:
      name: matcher
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "0.12.11"
  meta:
    dependency: transitive
    description:
      name: meta
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "1.7.0"
  oauth2:
    dependency: "direct main"
    description:
      name: oauth2
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "2.0.0"
  path:
    dependency: transitive
    description:
      name: path
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "1.8.0"
  path_provider:
    dependency: "direct main"
    description:
      name: path_provider
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "2.0.8"
  path_provider_android:
    dependency: transitive
    description:
      name: path_provider_android
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "2.0.9"
  path_provider_ios:
    dependency: transitive
    description:
      name: path_provider_ios
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "2.0.7"
  path_provider_linux:
    dependency: transitive
    description:
      name: path_provider_linux
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "2.1.4"
  path_provider_macos:
    dependency: transitive
    description:
      name: path_provider_macos
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "2.0.4"
  path_provider_platform_interface:
    dependency: transitive
    description:
      name: path_provider_platform_interface
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "2.0.1"
  path_provider_windows:
    dependency: transitive
    description:
      name: path_provider_windows
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "2.0.4"
  platform:
    dependency: transitive
    description:
      name: platform
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "3.1.0"
  plugin_platform_interface:
    dependency: transitive
    description:
      name: plugin_platform_interface
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "2.0.2"
  process:
    dependency: transitive
    description:
      name: process
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "4.2.4"
  sky_engine:
    dependency: transitive
    description: flutter
    source: sdk
    version: "0.0.99"
  source_span:
    dependency: transitive
    description:
      name: source_span
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "1.8.1"
  stack_trace:
    dependency: transitive
    description:
      name: stack_trace
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "1.10.0"
  stream_channel:
    dependency: transitive
    description:
      name: stream_channel
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "2.1.0"
  string_scanner:
    dependency: transitive
    description:
      name: string_scanner
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "1.1.0"
  term_glyph:
    dependency: transitive
    description:
      name: term_glyph
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "1.2.0"
  test_api:
    dependency: transitive
    description:
      name: test_api
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "0.4.3"
  typed_data:
    dependency: transitive
    description:
      name: typed_data
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "1.3.0"
  url_launcher:
    dependency: "direct main"
    description:
      name: url_launcher
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "6.0.17"
  url_launcher_android:
    dependency: transitive
    description:
      name: url_launcher_android
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "6.0.13"
  url_launcher_ios:
    dependency: transitive
    description:
      name: url_launcher_ios
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "6.0.13"
  url_launcher_linux:
    dependency: transitive
    description:
      name: url_launcher_linux
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "2.0.2"
  url_launcher_macos:
    dependency: transitive
    description:
      name: url_launcher_macos
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "2.0.2"
  url_launcher_platform_interface:
    dependency: transitive
    description:
      name: url_launcher_platform_interface
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "2.0.4"
  url_launcher_web:
    dependency: transitive
    description:
      name: url_launcher_web
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "2.0.5"
  url_launcher_windows:
    dependency: transitive
    description:
      name: url_launcher_windows
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "2.0.2"
  vector_math:
    dependency: transitive
    description:
      name: vector_math
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "2.1.1"
  win32:
    dependency: transitive
    description:
      name: win32
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "2.3.1"
  xdg_directories:
    dependency: transitive
    description:
      name: xdg_directories
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "0.2.0"
sdks:
  dart: ">=2.15.0 <3.0.0"
  flutter: ">=2.5.0"

```
</details>

**Comment by danagbemava-nc** [comment]
Hmmm, by all accounts, it should generate the `Podfile`. 

Can you try including the `desktop_windows` plugin to see if it generates the `Podfile`?

Also, please provide the output of `flutter run -v` (it may not help much in this case, but please provide it all the same).

**Comment by YowFung** [comment]
@danagbemava-nc 
> Can you try including the desktop_windows plugin to see if it generates the Podfile?

Still not working.

<details>
<summary>flutter run -d macos -v</summary>

```txt
❯ flutter run -d macos
Launching lib/main.dart on macOS in debug mode...
Exception: Podfile missing
❯ flutter run -d macos -v
[  +97 ms] executing: sysctl hw.optional.arm64
[  +18 ms] Exit code 1 from: sysctl hw.optional.arm64
[        ] sysctl: unknown oid 'hw.optional.arm64'
[   +6 ms] executing: [/Users/yowfung/Library/Flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +14 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] cf4400006550b70f28e4b4af815151d1e74846c6
[   +1 ms] executing: [/Users/yowfung/Library/Flutter/] git tag --points-at cf4400006550b70f28e4b4af815151d1e74846c6
[  +23 ms] Exit code 0 from: git tag --points-at cf4400006550b70f28e4b4af815151d1e74846c6
[        ] 2.8.0
[   +8 ms] executing: [/Users/yowfung/Library/Flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +12 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/stable
[        ] executing: [/Users/yowfung/Library/Flutter/] git ls-remote --get-url origin
[  +13 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +99 ms] executing: [/Users/yowfung/Library/Flutter/] git rev-parse --abbrev-ref HEAD
[  +15 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] stable
[   +6 ms] executing: sw_vers -productName
[  +17 ms] Exit code 0 from: sw_vers -productName
[   +1 ms] macOS
[        ] executing: sw_vers -productVersion
[  +18 ms] Exit code 0 from: sw_vers -productVersion
[        ] 12.0.1
[        ] executing: sw_vers -buildVersion
[  +17 ms] Exit code 0 from: sw_vers -buildVersion
[        ] 21A559
[  +76 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +4 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +52 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +31 ms] executing: sysctl hw.optional.arm64
[   +6 ms] Exit code 1 from: sysctl hw.optional.arm64
[        ] sysctl: unknown oid 'hw.optional.arm64'
[        ] executing: xcrun xcodebuild -version
[ +162 ms] Exit code 0 from: xcrun xcodebuild -version
[   +3 ms] Xcode 13.1
           Build version 13A1030d
[  +59 ms] Skipping pub get: version match.
[  +90 ms] Initializing file store
[  +11 ms] Skipping target: gen_localizations
[   +8 ms] gen_dart_plugin_registrant: Starting due to {InvalidatedReasonKind.inputChanged: The following inputs have updated contents:
/Users/yowfung/workspace/proitav/projects/TI_inventory_retrieval/.dart_tool/package_config_subset}
[  +30 ms] Found plugin desktop_window at /Users/yowfung/.pub-cache/hosted/pub.flutter-io.cn/desktop_window-0.4.0/
[  +30 ms] gen_dart_plugin_registrant: Complete
[   +1 ms] Skipping target: _composite
[   +2 ms] complete
[   +8 ms] Launching lib/main.dart on macOS in debug mode...
[   +6 ms] /Users/yowfung/Library/Flutter/bin/cache/dart-sdk/bin/dart --disable-dart-dev /Users/yowfung/Library/Flutter/bin/cache/artifacts/engine/darwin-x64/frontend_server.dart.snapshot --sdk-root
/Users/yowfung/Library/Flutter/bin/cache/artifacts/engine/common/flutter_patched_sdk/ --incremental --target=flutter --debugger-module-names --experimental-emit-debug-metadata -DFLUTTER_WEB_AUTO_DETECT=true --output-dill
/var/folders/vc/s675gvg95r5fwgh0yqxsh8k00000gn/T/flutter_tools.RuAKZt/flutter_tool.7qiTUu/app.dill --packages /Users/yowfung/workspace/proitav/projects/TI_inventory_retrieval/.dart_tool/package_config.json -Ddart.vm.profile=false
-Ddart.vm.product=false --enable-asserts --track-widget-creation --filesystem-scheme org-dartlang-root --initialize-from-dill build/c075001b96339384a97db4862b8ab8db.cache.dill.track.dill
--enable-experiment=alternative-invalidation-strategy
[  +63 ms] Found plugin desktop_window at /Users/yowfung/.pub-cache/hosted/pub.flutter-io.cn/desktop_window-0.4.0/
[  +56 ms] <- compile package:inventory_retrieval/main.dart
[+10873 ms] Exception: Podfile missing
[  +10 ms] "flutter run" took 11,564ms.
[   +5 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:674:9)
           <asynchronous suspension>
           #2      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1140:27)
           <asynchronous suspension>
           #3      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #4      CommandRunner.runCommand (package:args/command_runner.dart:209:13)
           <asynchronous suspension>
           #5      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:288:9)
           <asynchronous suspension>
           #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:236:5)
           <asynchronous suspension>
           #8      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #10     main (package:flutter_tools/executable.dart:94:3)
           <asynchronous suspension>
           
           
[ +257 ms] ensureAnalyticsSent: 255ms
[   +2 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1

```
</details>

**Comment by YowFung** [comment]
I deleted `pubspec.lock`, `.packages`, `.metadata`, `.flutter-plugins-dependencies`, `.flutter-plugins`, `build/`, `.dart_tool/`, `.idea/` and `macos/`, and then re-tried it. Still not working...

**Comment by danagbemava-nc** [comment]
Labelling for further investigation as I cannot reproduce this.

**Comment by YowFung** [comment]
I also had problems while compiling the application on Windows. It also seems couldn't create some files automatically. Is the same situation with above issue?

```txt
Launching lib\main.dart on Windows in debug mode...
Building Windows application...
CMake Error at CMakeLists.txt:51 (include):
  include could not find requested file:

    flutter/generated_plugins.cmake


Exception: Unable to generate build files
```

There is no `flutter/generated_plugins.cmake` file in `windows` folder.

<details>
<summary>flutter run -d windows -v</summary>

```txt
[  +58 ms] executing: [D:\SDK\Flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +53 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] cf4400006550b70f28e4b4af815151d1e74846c6
[        ] executing: [D:\SDK\Flutter/] git tag --points-at cf4400006550b70f28e4b4af815151d1e74846c6
[  +45 ms] Exit code 0 from: git tag --points-at cf4400006550b70f28e4b4af815151d1e74846c6
[        ] 2.8.0
[   +6 ms] executing: [D:\SDK\Flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +27 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/stable
[        ] executing: [D:\SDK\Flutter/] git ls-remote --get-url origin
[  +25 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +85 ms] executing: [D:\SDK\Flutter/] git rev-parse --abbrev-ref HEAD
[  +27 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] stable
[  +54 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +3 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +52 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +3 ms] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +67 ms] Skipping pub get: version match.
[  +68 ms] Initializing file store
[   +9 ms] Skipping target: gen_localizations
[   +5 ms] gen_dart_plugin_registrant: Starting due to {InvalidatedReasonKind.inputChanged: The following inputs have updated contents:
D:\workspace\git\ti_grab_goods\.dart_tool\package_config_subset,D:\workspace\git\ti_grab_goods\.dart_tool\flutter_build\generated_main.dart}
[  +38 ms] Found plugin path_provider at C:\Users\Zhanghuifeng\AppData\Local\Pub\Cache\hosted\pub.flutter-io.cn\path_provider-2.0.8\
[   +4 ms] Found plugin path_provider_android at C:\Users\Zhanghuifeng\AppData\Local\Pub\Cache\hosted\pub.flutter-io.cn\path_provider_android-2.0.9\
[   +1 ms] Found plugin path_provider_ios at C:\Users\Zhanghuifeng\AppData\Local\Pub\Cache\hosted\pub.flutter-io.cn\path_provider_ios-2.0.7\
[   +1 ms] Found plugin path_provider_linux at C:\Users\Zhanghuifeng\AppData\Local\Pub\Cache\hosted\pub.flutter-io.cn\path_provider_linux-2.1.4\
[   +1 ms] Found plugin path_provider_macos at C:\Users\Zhanghuifeng\AppData\Local\Pub\Cache\hosted\pub.flutter-io.cn\path_provider_macos-2.0.4\
[   +2 ms] Found plugin path_provider_windows at C:\Users\Zhanghuifeng\AppData\Local\Pub\Cache\hosted\pub.flutter-io.cn\path_provider_windows-2.0.4\
[  +11 ms] Found plugin url_launcher at C:\Users\Zhanghuifeng\AppData\Local\Pub\Cache\hosted\pub.flutter-io.cn\url_launcher-6.0.17\
[   +1 ms] Found plugin url_launcher_android at C:\Users\Zhanghuifeng\AppData\Local\Pub\Cache\hosted\pub.flutter-io.cn\url_launcher_android-6.0.13\
[   +1 ms] Found plugin url_launcher_ios at C:\Users\Zhanghuifeng\AppData\Local\Pub\Cache\hosted\pub.flutter-io.cn\url_launcher_ios-6.0.13\
[   +1 ms] Found plugin url_launcher_linux at C:\Users\Zhanghuifeng\AppData\Local\Pub\Cache\hosted\pub.flutter-io.cn\url_launcher_linux-2.0.2\
[   +1 ms] Found plugin url_launcher_macos at C:\Users\Zhanghuifeng\AppData\Local\Pub\Cache\hosted\pub.flutter-io.cn\url_launcher_macos-2.0.2\
[   +2 ms] Found plugin url_launcher_web at C:\Users\Zhanghuifeng\AppData\Local\Pub\Cache\hosted\pub.flutter-io.cn\url_launcher_web-2.0.5\
[   +1 ms] Found plugin url_launcher_windows at C:\Users\Zhanghuifeng\AppData\Local\Pub\Cache\hosted\pub.flutter-io.cn\url_launcher_windows-2.0.2\
[  +20 ms] gen_dart_plugin_registrant: Complete
[   +1 ms] Skipping target: _composite
[   +1 ms] complete
[   +5 ms] Launching lib\main.dart on Windows in debug mode...
[   +5 ms] D:\SDK\Flutter\bin\cache\dart-sdk\bin\dart.exe --disable-dart-dev D:\SDK\Flutter\bin\cache\artifacts\engine\windows-x64\frontend_server.dart.snapshot --sdk-root
D:\SDK\Flutter\bin\cache\artifacts\engine\common\flutter_patched_sdk/ --incremental --target=flutter --debugger-module-names --experimental-emit-debug-metadata -DFLUTTER_WEB_AUTO_DETECT=true --output-dill
C:\Users\ZHANGH~1\AppData\Local\Temp\flutter_tools.d6765f5\flutter_tool.b7091f2\app.dill --packages D:\workspace\git\ti_grab_goods\.dart_tool\package_config.json -Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts
--track-widget-creation --filesystem-scheme org-dartlang-root --initialize-from-dill build\c075001b96339384a97db4862b8ab8db.cache.dill.track.dill --enable-experiment=alternative-invalidation-strategy
[  +23 ms] executing: C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe -format json -products * -utf8 -latest -version 16 -requires Microsoft.VisualStudio.Workload.NativeDesktop
Microsoft.VisualStudio.Component.VC.Tools.x86.x64 Microsoft.VisualStudio.Component.VC.CMake.Project
[  +24 ms] Exit code 0 from: C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe -format json -products * -utf8 -latest -version 16 -requires Microsoft.VisualStudio.Workload.NativeDesktop
Microsoft.VisualStudio.Component.VC.Tools.x86.x64 Microsoft.VisualStudio.Component.VC.CMake.Project
[   +1 ms] [
             {
               "instanceId": "0d13811a",
               "installDate": "2021-02-03T01:25:07Z",
               "installationName": "VisualStudio/16.11.5+31729.503",
               "installationPath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community",
               "installationVersion": "16.11.31729.503",
               "productId": "Microsoft.VisualStudio.Product.Community",
               "productPath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\Common7\\IDE\\devenv.exe",
               "state": 4294967295,
               "isComplete": true,
               "isLaunchable": true,
               "isPrerelease": false,
               "isRebootRequired": false,
               "displayName": "Visual Studio Community 2019",
               "description": "Powerful IDE, free for students, open-source contributors, and individuals",
               "channelId": "VisualStudio.16.Release",
               "channelUri": "https://aka.ms/vs/16/release/channel",
               "enginePath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\resources\\app\\ServiceHub\\Services\\Microsoft.VisualStudio.Setup.Service",
               "releaseNotes": "https://docs.microsoft.com/en-us/visualstudio/releases/2019/release-notes-v16.11#16.11.5",
               "thirdPartyNotices": "https://go.microsoft.com/fwlink/?LinkId=660909",
               "updateDate": "2021-10-15T08:19:18.4650492Z",
               "catalog": {
                 "buildBranch": "d16.11",
                 "buildVersion": "16.11.31729.503",
                 "id": "VisualStudio/16.11.5+31729.503",
                 "localBuild": "build-lab",
                 "manifestName": "VisualStudio",
                 "manifestType": "installer",
                 "productDisplayVersion": "16.11.5",
                 "productLine": "Dev16",
                 "productLineVersion": "2019",
                 "productMilestone": "RTW",
                 "productMilestoneIsPreRelease": "False",
                 "productName": "Visual Studio",
                 "productPatchVersion": "5",
                 "productPreReleaseMilestoneSuffix": "1.0",
                 "productSemanticVersion": "16.11.5+31729.503",
                 "requiredEngineVersion": "2.11.40.25675"
               },
               "properties": {
                 "campaignId": "44450884.1594276487",
                 "channelManifestId": "VisualStudio.16.Release/16.11.5+31729.503",
                 "nickname": "",
                 "setupEngineFilePath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\vs_installershell.exe"
               }
             }
           ]
[   +3 ms] Building Windows application...
[  +28 ms] <- compile file:///D:/workspace/git/ti_grab_goods/.dart_tool/flutter_build/generated_main.dart
[   +2 ms] executing: C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe -S D:\workspace\git\ti_grab_goods\windows -B build\windows -G Visual Studio 16 2019
[ +118 ms] CMake Error at CMakeLists.txt:51 (include):
[   +2 ms]   include could not find requested file:
[        ]     flutter/generated_plugins.cmake
[   +1 ms] -- Configuring incomplete, errors occurred!
[        ] See also "D:/workspace/git/ti_grab_goods/build/windows/CMakeFiles/CMakeOutput.log".
[  +11 ms] Building Windows application... (completed in 158ms)
[+8238 ms] Exception: Unable to generate build files
[   +1 ms] "flutter run" took 8,808ms.
[   +3 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:674:9)
           <asynchronous suspension>
           #2      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1140:27)
           <asynchronous suspension>
           #3      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #4      CommandRunner.runCommand (package:args/command_runner.dart:209:13)
           <asynchronous suspension>
           #5      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:288:9)
           <asynchronous suspension>
           #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:236:5)
           <asynchronous suspension>
           #8      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #10     main (package:flutter_tools/executable.dart:94:3)
           <asynchronous suspension>
           
           
[ +257 ms] ensureAnalyticsSent: 253ms
[   +2 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1
```
</details>

<details>
<summary>flutter doctor -v</summary>

```txt
[√] Flutter (Channel stable, 2.8.0, on Microsoft Windows [Version 10.0.17763.2300], locale
    en-US)
    • Flutter version 2.8.0 at D:\SDK\Flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • FLUTTER_GIT_URL = https://github.com/flutter/flutter.git
    • Framework revision cf44000065 (7 days ago), 2021-12-08 14:06:50 -0800
    • Engine revision 40a99c5951
    • Dart version 2.15.0
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at D:\SDK\Android
    • Platform android-31, build-tools 30.0.3
    • ANDROID_HOME = D:\SDK\Android
    • ANDROID_SDK_ROOT = D:\SDK\Android
    • Java binary at: C:\Program Files\Java\jdk1.8.0_251\bin\java
    • Java version Java(TM) SE Runtime Environment (build 1.8.0_251-b08)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • CHROME_EXECUTABLE = C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.5)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31729.503
    • Windows 10 SDK version 10.0.19041.0

[!] Android Studio (not installed)
    • Android Studio not found; download from
      https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for
      detailed instructions).

[√] IntelliJ IDEA Community Edition (version 2021.2)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2021.2.3
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5486

[√] VS Code (version 1.55.2)
    • VS Code at C:\Users\Zhanghuifeng\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.15.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version
      10.0.17763.2300]
    • Chrome (web)      • chrome  • web-javascript • unknown
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 95.0.1020.53

! Doctor found issues in 1 category.
```
</details>

**Comment by danagbemava-nc** [comment]
A similar issue is being tracked in https://github.com/flutter/flutter/issues/94874

**Comment by YowFung** [comment]
I think I've figured out the reproduction pattern of this issue.

Add `example` folder in the root directory of your project, and the issue occurred. Like the following:

![image](https://user-images.githubusercontent.com/26346560/147059337-3f047c25-5f68-40cd-81c6-ab92b83bbfc4.png)

Note that the `example` folder is not the folder of the demo app of flutter, it just a general folder.

**Comment by stuartmorgan** [comment]
Given the last comment, this is a duplicate of https://github.com/flutter/flutter/issues/87007
